### ISSUE TITLE
An option to export CSV file with 10,000 rows limitation

### ISSUE THREAD

**Issue by zhongnansu** [opensearch-project/dashboards-reporting#60]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context or screenshots about the feature request here

--- 

Is it possible to have a feature to allow user to download CSV without maximum of 10,000 rows limitation? Can someone give me some hints how to implement such feature? Where will be the best place (code file) to add a checkbox to enable such feature? Where will be the best place I can utilize "scroll" feature to retrieve all the data? I am not very familiar with ES and Kibana source code layout. But eager to have such feature implemented. Thanks!

This issue is submitted by @FromPAUS
https://github.com/opendistro-for-elasticsearch/kibana-reports/issues/176

**Comment by mhoydis13** [comment]
> 

Looks like this file has moved to here: 
https://github.com/opensearch-project/dashboards-reporting/blob/main/public/components/report_definitions/report_settings/report_settings.tsx#L172

Can you provide more details on how a user would implement this workaround?  What would we have to do to make this change and have it take effect?  Does it require a custom build of the dashboards-reports plugin?

**Comment by mhoydis13** [comment]
> You can increase the CSV records limit at https://github.com/opensearch-project/dashboards-reports/blob/main/dashboards-reports/public/components/report_definitions/report_settings/report_settings.tsx#L172. Make sure you recreate any existing report definitions in OSD or this change won't be picked up.

Looks like this file has moved to here:
https://github.com/opensearch-project/dashboards-reporting/blob/main/public/components/report_definitions/report_settings/report_settings.tsx#L172

Can you provide more details on how a user would implement this workaround? What would we have to do to make this change and have it take effect? Does it require a custom build of the dashboards-reports plugin?

**Comment by ProfConsulting** [comment]
Waiting for this, too.

**Comment by guillaumeldc** [comment]
Waiting for this feature as well

**Comment by CauchY75** [comment]
we are looking this solution as well.

**Comment by XandorMaker** [comment]
> Reference in new

Was you able to figure it out?

**Comment by Beastofshado** [comment]
Hi everyone, 

I fell into the same problem, I've a wazuh single docker environment and  after checking the server side I found this:

on the docker single-node-wazuh.indexer-1 where is installed OpenSearch I was able to found this file called: "costants".js where is possible to found the variable **DEFAULT_MAX_SIZE**, I've changed the value on this variable and restarted all the wazuh dockers.
I tested and then now I'm able to create reports and export as .csv and the file generated passed that of 10k limits rows that is putted by default.

Here more details:

Wazuh version 4.7.0
OpenSearch 7.10.2
Stack management 2.8.0

How to:

1. Modify the costants.js file 

`nano /usr/share/wazuh-dashboard/plugins/reportsDashboards/server/routes/utils/constants.js`

2. Search the following string: **DEFAULT_MAX_SIZE =**

3. Verify that this match with this line:

`const DEFAULT_MAX_SIZE = 10000`

4. Increase/Change the value on what you prefer i.e. of new value:

`const DEFAULT_MAX_SIZE = 999999`

5. Restart all the services of wazuh 


N.B.: I don't know if the file is the same when you install wazuh on a on-premises server with OpenSearch.
